src/server/SlotMachine.ts

```diff
@@ -0,0 +1,253 @@
+      case 7:
+        symbolsArray.push([4, 1, 1, 1, 2]);
+        symbolsArray.push([1, 0, 2, 3, 1]);
+        symbolsArray.push([3, 2, 0, 2, 3]);
+        break;
+    }
+
+    return symbolsArray;
+  };
+
+  private calculatePaylines = (symbolsArray: number[][], stake: number) => {
+    const gottenPaylinesInfo: GottenPaylineInfo[] = [];
+
+    paylines.forEach((payline: Payline) => {
+      const actualLine: number[] = [];
+      const paylineArray = payline.payline;
+
+      for (let i = 0; i < this.numberOfStrips; i++) {
+        for (let j = 0; j < this.numberOfRowsAsResult; j++) {
+          //console.log("paylineArray[j]", paylineArray[j]);
+          //console.log("paylineArray[j][i]", paylineArray[j][i]);
+          if (paylineArray[j][i]) {
+            actualLine.push(symbolsArray[j][i]);
+          }
+        }
+      }
+
+      let numConsecutiveCoincidences = 1;
+      let numBigWins = 0;
+      let idFirstSymbol = actualLine.find(
+        (id) => id != wildcardSymbol.id && id != wildcardBigWinSymbol.id
+      );
+
+      if (idFirstSymbol != undefined) {
+        for (let i = 1; i < actualLine.length; i++) {
+          if (
+            idFirstSymbol == actualLine[i] ||
+            actualLine[i] == wildcardSymbol.id
+          ) {
+            numConsecutiveCoincidences++;
+          } else if (actualLine[i] == wildcardBigWinSymbol.id) {
+            numConsecutiveCoincidences++;
+            numBigWins++;
+          } else {
+            break;
+          }
+        }
+      } else {
+        numConsecutiveCoincidences = 5;
+        numBigWins = actualLine.reduce(
+          (total, id) => (id == wildcardBigWinSymbol.id ? total++ : total),
+          0
+        );
+      }
+
+      if (numConsecutiveCoincidences >= 3) {
+        const amountWinByLine =
+          //stake +
+          stake * this.winPercentageBySymbol * numConsecutiveCoincidences;
+        let gottenPaylineInfo = new GottenPaylineInfo(
+          payline,
+          numConsecutiveCoincidences,
+          numBigWins,
+          amountWinByLine,
+          stake
+        );
+        gottenPaylinesInfo.push(gottenPaylineInfo);
+      }
+    });
+
+    return gottenPaylinesInfo;
+  };
+
+  private calculateAmountTotalWin = (
+    gottenPaylinesInfo: GottenPaylineInfo[]
+  ) => {
+    let amountTotalWin = gottenPaylinesInfo.reduce(
+      (total, paylineInfo: GottenPaylineInfo) =>
+        total + paylineInfo.amountWinByLine,
+      0
+    );
+
+    amountTotalWin = Math.round((amountTotalWin + Number.EPSILON) * 100) / 100;
+
+    if (amountTotalWin > 0) {
+      amountTotalWin += gottenPaylinesInfo[0].stake;
+    }
+
+    return amountTotalWin;
+  };
+
+  initSymbols = (idRequest: number) => {
+    const symbolsArray = this.generateSymbolsArray();
+
+    return {
+      idRequest: idRequest,
+      symbolsArray: symbolsArray,
+    };
+  };
+
+  spin = (
+    stake: number,
+    idRequest: number,
+    idCheat: number | undefined = undefined
+  ) => {
```

Comment:
function signature is missing the return type. Typescript is smart enough to understand return type on the fly so usually it is not necessary, but in that case, you already have declared type "SpinResponse" which you can use here as the return type.